src/main/java/com/endava/soa_5/actions/GenericActions.java

```diff
@@ -1,12 +1,43 @@
 package com.endava.soa_5.actions;
 
-import com.endava.soa_5.abstract_classes.BaseAction;
+import com.endava.soa_5.abstract_classes.BaseClass;
+
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.MatcherAssert.*;
+import static org.hamcrest.core.Is.is;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.FluentWait;
+import org.openqa.selenium.support.ui.Wait;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.NoSuchElementException;
 
-public class GenericActions extends BaseAction {
-    public static void clickOnElement(WebElement element){
+public class GenericActions extends BaseClass {
+    Logger log = LoggerFactory.getLogger(this.getClass());
+
+    public void clickOnElement(WebElement element){
         element.click();
     }
-    //TODO: add generic waiter method here;
-    //TODO: add some generic asserts here;
-}
+
+    public void waitForElement(WebElement element, int timeout, int checkFrequency) {
```

Comment:
timeout and checkFrequency can be added in property file and can be removed from method arguments

Reply:
will be added in later PR's

Reply:
done